aptos-move/block-executor/src/executor.rs

```diff
@@ -216,16 +227,48 @@ where
 
         let _timer = WORK_WITH_TASK_SECONDS.start_timer();
         let mut scheduler_task = SchedulerTask::NoTask;
+        let mut accumulated_gas = 0;
         loop {
             // Only one thread try_commit to avoid contention.
             if committing {
                 // Keep committing txns until there is no more that can be committed now.
                 while let Some(txn_idx) = scheduler.try_commit() {
+                    // Committed the last transaction, BlockSTM finishes execution.
                     if txn_idx as usize + 1 == block.len() {
-                        // Committed the last transaction / everything.
                         scheduler_task = SchedulerTask::Done;
                         break;
                     }
+
+                    // When there is a txn with Abort or SkipRest as the execution output, can early halt BlockSTM.
+                    if let ExecutionStatus::Abort(_) | ExecutionStatus::SkipRest(_) =
```

Comment:
I'd make the function check the status, as there seems no need to return the write_set and then just check matches?
i.e. if last_input_output.skipped_or_aborted(txn_idx) {
}

Reply:
Alternatively, you could have the function last_input_output.success_gas(txn_idx) that returns Ok(txn_gas) when it is Success and anyhow::Err otherwise, so you can just match it, and either halt/break or add to accumulated gas.